examples/tutorials/online_asr_tutorial.py

```diff
@@ -0,0 +1,160 @@
+# Streaming inference works on input data with overlap.
+# Emformer RNN-T expects right context like illustrated like the
+# following.
+#
+# .. image:: https://download.pytorch.org/torchaudio/tutorial-assets/emformer_rnnt_context.png
+#
+# The size of main segment and right context can be retrieved from bundle.
+#
+
+frames_per_chunk = bundle.segment_length * bundle.hop_length
+right_context_size = bundle.right_context_length * bundle.hop_length
+
+print(f'Main segment size: {frames_per_chunk} frames ({frames_per_chunk / bundle.sample_rate} seconds)')
+print(f'Right context size: {right_context_size} frames ({right_context_size / bundle.sample_rate} seconds)')
+
+######################################################################
+# 4. Configure the audio stream
+# -----------------------------
+#
+# Next, we configure the input audio stream using :py:func:`~torchaudio.prototype.io.Streamer`.
+#
+# For the detail of this API, please refer to the
+# `Media Stream API tutorial <../tutorials/streaming_api_tutorial.html>`__.
+#
+
+src = (
+    "https://ia800302.us.archive.org/33/items/"
+    "great_pirate_stories_0905_librivox.org/greatpiratestories_00_various.mp3"
+)
+
+streamer = Streamer(src)
+streamer.add_basic_audio_stream(
+    frames_per_chunk=frames_per_chunk, sample_rate=bundle.sample_rate)
+
+print(streamer.get_src_stream_info(0))
+print(streamer.get_out_stream_info(0))
+
+######################################################################
+# `Streamer` does not provide right context, so we create a wrapper
+# that caches the segment and append right context.
+
+
+class ContextStream:
+    def __init__(self, streamer, right_context):
+        self.streamer = streamer
+        self.right_context = right_context
+
+    def __iter__(self):
+        streamer = self.streamer.stream()
+        segment, = next(streamer)
+        for next_segment, in streamer:
+            chunk = torch.cat((segment, next_segment[:self.right_context, :]))
+            segment = next_segment
+            yield chunk.T[0]
+
+
+######################################################################
+# The following is the audio we will stream.
+#
+# .. raw:: html
+#
+#    <audio controls>
+#       <source src="https://ia800302.us.archive.org/33/items/great_pirate_stories_0905_librivox.org/greatpiratestories_00_various.mp3" type="audio/mp3">
+#    Your browser does not support the audio element.
+#    </audio>
+#
+
+######################################################################
+# 5. Run stream inference
+# -----------------------
+#
+# Finally, we run the recognition.
+#
+
+context_streamer = ContextStream(streamer, right_context_size)
+state, hypothesis = None, None
+
+with torch.no_grad():
+    for i, chunk in enumerate(context_streamer):
+        features, length = feature_extractor(chunk)
+        hypos, state = decoder.infer(features, length, 10, state=state, hypothesis=hypothesis)
+        hypothesis = hypos[0]
```

Comment:
@hwangjeff I am observing a surprising result. When I remove `hypothesis = hypos[0]` and pass `hypos[0].tokens` directly to `token_processor`, then the result contains duplications. Do you know why?

### With `hypothesis = hypos[0]`

<img width="766" alt="Screen Shot 2022-02-02 at 3 53 59 PM" src="https://user-images.githubusercontent.com/855818/152107915-df192658-375e-47a5-9c11-480bcceb0bab.png">

### Without `hypothesis = hypos[0]`

<img width="768" alt="Screen Shot 2022-02-02 at 3 55 59 PM" src="https://user-images.githubusercontent.com/855818/152107886-339a64f0-e427-4c7e-a2fb-b80ff631d1fe.png">

Reply:
this is expected. `hypothesis` is actually module-scope, so in the first case, we save each decoder call's top hypothesis to pass to the next decoder call, whereas in the second we are always passing in `hypothesis=None`. each generated `hypothesis` contains the last predicted word piece and last prediction network (language model) state. without these, the model would be primarily relying on acoustic data to make its prediction, i.e. it wouldn't be aware of what word pieces it had previously predicted, and therefore it would be more prone to predicting duplicate pieces

Reply:
ah okay, I missed the fact that hypothesis needs to be updated. Thanks for the explanation!